cli/cmd/install.go

```diff
@@ -90,254 +72,111 @@ var (
-
-/* The install commands all follow the same flow:
- * 1. Load default values from the Linkerd2 chart
- * 2. Apply flags to modify the values
- * 3. Render the chart using those values
- *
- * The individual commands (install, install config, and install control-plane)
- * differ in which flags are available to each, what pre-check validations
- * are done, and which subset of the chart is rendered.
- */
-
-func newCmdInstallConfig(values *l5dcharts.Values) *cobra.Command {
-	flags, flagSet := makeAllStageFlags(values)
+func newCmdInstall() *cobra.Command {
+	values, err := l5dcharts.NewValues()
+	if err != nil {
+		fmt.Fprintln(os.Stderr, err.Error())
+		os.Exit(1)
+	}
+	var crds bool
 	var options valuespkg.Options
 
+	installOnlyFlags, installOnlyFlagSet := makeInstallFlags(values)
+	installUpgradeFlags, installUpgradeFlagSet, err := makeInstallUpgradeFlags(values)
+	if err != nil {
+		fmt.Fprint(os.Stderr, err.Error())
+		os.Exit(1)
+	}
+	proxyFlags, proxyFlagSet := makeProxyFlags(values)
+
+	flags := flattenFlags(installOnlyFlags, installUpgradeFlags, proxyFlags)
+
 	cmd := &cobra.Command{
-		Use:   "config [flags]",
+		Use:   "install [flags]",
 		Args:  cobra.NoArgs,
-		Short: "Output Kubernetes cluster-wide resources to install Linkerd",
-		Long: `Output Kubernetes cluster-wide resources to install Linkerd.
+		Short: "Output Kubernetes configs to install Linkerd",
+		Long: `Output Kubernetes configs to install Linkerd.
 
-This command provides Kubernetes configs necessary to install cluster-wide
-resources for the Linkerd control plane. This command should be followed by
-"linkerd install control-plane".`,
+This command provides all Kubernetes configs necessary to install the Linkerd
+control plane.`,
 		Example: `  # Default install.
-  linkerd install config | kubectl apply -f -
+  linkerd install | kubectl apply -f -
 
   # Install Linkerd into a non-default namespace.
-  linkerd install config -L linkerdtest | kubectl apply -f -
+  linkerd install -l linkerdtest | kubectl apply -f -
 
 The installation can be configured by using the --set, --values, --set-string and --set-file flags.
 A full list of configurable values can be found at https://www.github.com/linkerd/linkerd2/tree/main/charts/linkerd2/README.md`,
 		RunE: func(cmd *cobra.Command, args []string) error {
-			err := flag.ApplySetFlags(values, flags)
-			if err != nil {
-				return err
-			}
-
 			// Create values override
 			valuesOverrides, err := options.MergeValues(nil)
 			if err != nil {
 				return err
 			}
-
+			var k8sAPI *k8s.KubernetesAPI
 			if !ignoreCluster {
-				// Ensure k8s is reachable and that Linkerd is not already installed.
+				// Ensure k8s is reachable
 				if err := errAfterRunningChecks(values.CNIEnabled); err != nil {
 					if healthcheck.IsCategoryError(err, healthcheck.KubernetesAPIChecks) {
 						fmt.Fprintf(os.Stderr, errMsgCannotInitializeClient, err)
-					} else {
-						fmt.Fprintf(os.Stderr, errMsgGlobalResourcesExist, err)
```

Comment:
With this removal, aren't we swallowing errors that are not `healthcheck.KubernetesAPIChecks`?